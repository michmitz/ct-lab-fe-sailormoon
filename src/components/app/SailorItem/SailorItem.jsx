import React from 'react';
import PropTypes from 'prop-types';

const SailorItem = ({ sailorName, imageUrl }) => {
  return <figure>
    <img src={imageUrl} alt={sailorName} />
    <figcaption>{sailorName}</figcaption>
  </figure>
} 

SailorItem.propTypes = {
  sailorName: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired
}

export default SailorItem
