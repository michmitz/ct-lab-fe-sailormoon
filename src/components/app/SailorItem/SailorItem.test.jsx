import React from 'react';
import { render, cleanup } from '@testing-library/react';
import SailorItem from './SailorItem';

describe('CharacterItem component', () => {
  afterEach(() => cleanup());

  it('renders single character', () => {
    const { asFragment } = render(<SailorItem
      sailorName='Sailor Michelle'
      imageUrl='url@url.com'
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
