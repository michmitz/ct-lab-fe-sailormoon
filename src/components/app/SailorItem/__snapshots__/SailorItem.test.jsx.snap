// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterItem component renders single character 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="Sailor Michelle"
      src="url@url.com"
    />
    <figcaption>
      Sailor Michelle
    </figcaption>
  </figure>
</DocumentFragment>
`;
